### PR TITLE
Basic geometric transform functions.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

#### Why we need this PR?
- adds functions that generate common affine transform matrices

#### What happend in this PR?
Added functions for
- translation (2D, 3D)
- non-uniform scaling along major axes (2D, 3D)
- shear (2D, 3D)
- rotation (2D)
All transforms are homogeneous (i.e. can be extended to perspective transforms!)

### CODE REVIEW

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

File: include/dali/core/geom/transform.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+#ifndef DALI_CORE_GEOM_TRANSFORM_H_
+#define DALI_CORE_GEOM_TRANSFORM_H_
+
+#include "dali/core/geom/mat.h"
+
+namespace dali {
+
+DALI_HOST_DEV
+constexpr mat3 translation(vec2 offset) {
```
Comment (klecki) [inline include/dali/core/geom/transform.h]: Why no templates? I can imagine integer translation or operations with higher precision.

Comment (mzient) [reply include/dali/core/geom/transform.h]: Integer transforms make little sense. I was thinking about it in the context of higher precision. This variant won for being able to accept braced initializers without ambiguity.

**Pull request merged**